src/bin/cargo/commands/yank.rs

```diff
@@ -28,14 +27,37 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
 
     let registry = args.registry(config)?;
 
+    let (krate, version) = resolve_crate(args.value_of("crate"), args.value_of("version"))?;
+    if version.is_none() {
+        return Err(anyhow::format_err!("`--version` is required").into());
+    }
+
     ops::yank(
         config,
-        args.value_of("crate").map(|s| s.to_string()),
-        args.value_of("version").map(|s| s.to_string()),
+        krate.map(|s| s.to_string()),
```

Comment:
Could we remove the `.map(|s| s.to_string())` for all these arguments and make `ops::yank` take `Option<&str>` instead?

Reply:
imo
- this is out of scope for this PR
- the use of `String` is wider than just `yank`, its most Op options (though sometimes we do see a `&str` or a mix of `String` and `&str`)
- I personally find this a point I don't have a strong preference on besides consistency.  `String` is slightly more ergonomic though that is diminished by having a `&Config` in the `Options` and no one will ever notice the performance difference in these (thinking back to the RustConf talk about "just clone and stop worrying about it).